contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/heartbeat_metrics/heartbeat.py

```diff
@@ -12,22 +12,59 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import datetime
+import json
 import os
 import platform
 from collections import OrderedDict
 
+import requests
+
 from opencensus.common.version import __version__ as opencensus_version
 from opencensus.ext.azure.common.version import __version__ as ext_version
 from opencensus.metrics.export.gauge import LongGauge
 from opencensus.metrics.label_key import LabelKey
 from opencensus.metrics.label_value import LabelValue
 
+_AIMS_URI = "http://169.254.169.254/metadata/instance/compute"
+_AIMS_API_VERSION = "api-version=2017-12-01"
+_AIMS_FORMAT = "format=json"
+
 
 class HeartbeatMetric:
     NAME = "Heartbeat"
 
     def __init__(self):
+        self.vm_data = {}
+        self.is_vm = False
         self.properties = OrderedDict()
+        self.update_properties()
+        self.heartbeat = LongGauge(
+            HeartbeatMetric.NAME,
+            'Heartbeat metric with custom dimensions',
+            'count',
+            list(self.properties.keys()),
+        )
+        self.heartbeat.get_or_create_time_series(
+            list(self.properties.values())
+        )
+
+    def get_metrics(self):
+        if self.is_vm:
+            # Only need to update if in vm (properties could change)
+            self.properties.clear()
+            self.update_properties()
+            self.heartbeat.clear()
+            # pylint: disable=protected-access
```

Comment:
Is no other way to update the label keys value?, maybe creating the LongGauge again?

Reply:
I was thinking that maybe this would eventually take up too much memory, but Python gc is pretty efficient so we can do it this way.